### PR TITLE
Add guides for labels and annotations

### PR DESCRIPTION
**Pull request by zli82016**
Add the introduction to the new labels and annotations models and will add more details later.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Review by c2thorn — CHANGES_REQUESTED** [review]
This is a provider-level change, so please but it slightly lower in the 

## Provider section

You can then remove the example where it says "Provider-level change example header"

**Comment by zli82016** [comment]
Thanks so much for the comments.

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 1]
```diff
@@ -86,6 +86,28 @@ terraform {
 }
 ```
 
+## Lables Rework
+
+### Labels
+
+The new labels model will be applied to all of the resources with the resource `labels` field. Some `labels` fields are for child resources, so the new model will not be applied to the `labels` fields for child resources.
+
+The default labels configured on the provider through the new `default_lables` field will be supported. The default labels configured on the provider will be applied to all of the resources with the resource `labels` field.
```
[the remark this answers is not in the document]
Comment (zli82016) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown]: Added a separate section for provider default labels

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 2]
```diff
@@ -86,6 +86,28 @@ terraform {
 }
 ```
 
+## Lables Rework
+
+### Labels
+
+The new labels model will be applied to all of the resources with the resource `labels` field. Some `labels` fields are for child resources, so the new model will not be applied to the `labels` fields for child resources.
```
[the remark this answers is not in the document]
Comment (zli82016) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown]: Added more information.

**Review by c2thorn — COMMENTED** [review]
Section looks much better, thank you for adding clarifications! Can we move the entire section below the ## Provider line on line 119? Then delete the

### Provider-level change example header
Description of the change and how users should adjust their configuration (if needed).

lines?

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 3]
```diff
@@ -88,9 +88,35 @@ terraform {
 
 ## Provider
 
-### Provider-level change example header
+## Provider-level Labels Rework
```
Comment (zli82016) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:91]: Is Ok to use "##" for Provider-level Labels Rework ?

**Comment by zli82016** [comment]
> Section looks much better, thank you for adding clarifications! Can we move the entire section below the ## Provider line on line 119? Then delete the
> 
> ### Provider-level change example header
> Description of the change and how users should adjust their configuration (if needed).
> 
> lines?

Done.

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown (hunk 3 above)
Comment (c2thorn) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:91]: I would prefer it to be under the ## Provider section, so one step lower at "###". I guess that would mean all of the subsections need to have an extra "#" as well. Sorry.

Comment (zli82016) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown]: Made the changes. Thanks.

**Review by c2thorn — APPROVED** [review]

**Pull request merged**